### PR TITLE
Closes #2706: Do not remember site permissions in private browsing.

### LINKED ISSUE #2706: Do not remember site permissions in private browsing
Related issue: https://github.com/mozilla-mobile/fenix/issues/1309

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 1]
```diff
@@ -120,18 +120,23 @@ class SitePermissionsFeature(
...
 
     @Synchronized internal fun storeSitePermissions(
+        session: Session,
         request: PermissionRequest,
         permissions: List<Permission> = request.permissions,
         status: SitePermissions.Status
     ) {
+        if (session.private) {
+            return
+        }
```
Comment (pocmo) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: I wonder if we should do that check one level higher where we also check `shouldStore` .. there's probably no need to launch the coroutine?

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 2]
```diff
@@ -176,8 +181,12 @@ class SitePermissionsFeature(
             return null
         }
 
-        val permissionFromStorage = withContext(ioCoroutineScope.coroutineContext) {
-            storage.findSitePermissionsBy(request.host)
+        val permissionFromStorage = if (session.private) {
+            null
+        } else {
```
Comment (pocmo) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Is the opposite really true too? On desktop permissions granted in normal browsing seem to be granted in private browsing automatically too.

Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: You right, I will remove this one!

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Yeah, that could be much better. I will add the coroutine launch in `storeSitePermissions` function!

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 3]
```diff
@@ -68,14 +69,19 @@ class SitePermissionsFeature(
...
+        CoroutineScope(Dispatchers.IO)
+    }
 
     override fun start() {
         observer.observeSelected()
     }
 
     override fun stop() {
         observer.stop()
+        ioCoroutineScope.cancel()
```
Comment (Amejia481) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: @pocmo I also added a this to cancel all the coroutines when the feature is stopped

**Comment by Amejia481** [comment]
@pocmo PR updated :)

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt (hunk 3 above)
Comment (pocmo) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Doesn't this mean the feature won't work anymore on the next start? Or can you continue to use the scope after cancelling?

Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Good point, I will do some research!

Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: I will address that in https://github.com/mozilla-mobile/android-components/issues/2889. I will remove this part from this PR

Comment (pocmo) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: 👍

**Pull request merged**